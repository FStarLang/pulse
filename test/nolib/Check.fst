module Check

#lang-pulse
open Pulse.Nolib

ghost
fn test (x : int)
  requires pure (x == 'y)
  returns z : int
  ensures pure (z == x)
{
  x
}

#check test
