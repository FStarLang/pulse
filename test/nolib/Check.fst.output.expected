* Info at Check.fst(15,0-15,11):
  - Term
    Check.test
    has type
    x: Prims.int
      -> Pulse.Lib.Core.stt_ghost Prims.int
          Pulse.Lib.Core.emp_inames
          (Pulse.Lib.Core.pure (x == 'y))
          (fun x -> Pulse.Lib.Core.pure (x == x))

